main/src/ca/uwaterloo/flix/language/ast/SyntaxTree.scala

```diff
@@ -0,0 +1,504 @@
+
+    case object Kind extends TreeKind
+
+    case object Modifiers extends TreeKind
+
+    case object Operator extends TreeKind
+
+    case object Parameter extends TreeKind
+
+    case object Parameters extends TreeKind
+
+    case object QName extends TreeKind
+
+    case object Source extends TreeKind
+
+    case object TypeParameter extends TreeKind
+
+    case object TypeParameters extends TreeKind
+
+    /////////////// DECLARATIONS ///////////////
+    sealed trait Decl extends TreeKind {
+      override def debug_name: Option[String] = Some("Decl")
+    }
+
+    object Decl {
+      case object AssociatedTypeDef extends Decl
+
+      case object AssociatedTypeSig extends Decl
+
+      case object Class extends Decl
+
+      case object Def extends Decl
+
+      case object Effect extends Decl
+
+      case object Enum extends Decl
+
+      case object Instance extends Decl
+
+      case object Law extends Decl
+
+      case object Module extends Decl
+
+      case object Operation extends Decl
+
+      case object RestrictableEnum extends Decl
+
+      case object Signature extends Decl
+
+      case object TypeAlias extends Decl
+    }
+
+    /////////////// EXPRESSIONS  ///////////////
+    sealed trait Expr extends TreeKind {
+      override def debug_name: Option[String] = Some("Expr")
+    }
+
+    object Expr {
+
+      /**
+       * A marker kind used to wrap nested expressions.
+       * For instance on a binary expression "1 + 2" you would do
+       * Expr
+       *   Binary
+       *     Expr
+       *       LiteralNumber
+       *     Operator
+       *     Expr
+       *        LiteralNumber
+       */
+      case object Expr extends Expr
+
+      case object Ascribe extends Expr
+
+      case object Binary extends Expr
+
+      case object Block extends Expr
+
+      case object Call extends Expr
+
+      case object CheckedEffectCast extends Expr
+
+      case object CheckedTypeCast extends Expr
+
+      case object Do extends Expr
+
+      case object FixpointConstraint extends Expr
+
+      case object FixpointConstraintSet extends Expr
+
+      case object FixpointFrom extends Expr
+
+      case object FixpointInject extends Expr
+
+      case object FixpointQuery extends Expr
+
+      case object FixpointSelect extends Expr
+
+      case object FixpointSolve extends Expr
+
+      case object FixpointWhere extends Expr
+
+      case object ForApplicative extends Expr
+
+      case object Foreach extends Expr
+
+      case object ForeachYield extends Expr
+
+      case object ForMonadic extends Expr
+
+      case object Generator extends Expr
+
+      case object Guard extends Expr
+
+      case object Hole extends Expr
+
+      case object HoleVariable extends Expr
+
+      case object IfThenElse extends Expr
+
+      case object InstanceOf extends Expr
+
+      case object Intrinsic extends Expr
+
+      case object JvmMethod extends Expr
+
+      case object KeyValue extends Expr
```

Comment:
This is for Map literals?

Reply:
Exacly. I intentionally named it very generally because I though I would be able to reuse it, but that didn't turn out.

Reply:
I renamed it `LiteralMapKeyValueFragment`